src/services/jsDoc.ts

```diff
@@ -92,9 +92,16 @@ namespace ts.JsDoc {
         // from Array<T> - Array<string> and Array<number>
         const parts: SymbolDisplayPart[][] = [];
         forEachUnique(declarations, declaration => {
-            for (const { comment } of getCommentHavingNodes(declaration)) {
-                if (comment === undefined) continue;
-                const newparts = getDisplayPartsFromComment(comment, checker);
+            for (const jsdoc of getCommentHavingNodes(declaration)) {
+                if (jsdoc.comment === undefined
+                    || isJSDoc(jsdoc)
+                       && declaration.kind !== SyntaxKind.JSDocTypedefTag
+                       && jsdoc.tags
+                       && jsdoc.tags.some(t => t.kind === SyntaxKind.JSDocTypedefTag || t.kind === SyntaxKind.JSDocCallbackTag)
+                       && !jsdoc.tags.some(t => t.kind === SyntaxKind.JSDocParameterTag || t.kind === SyntaxKind.JSDocReturnTag)) {

```

Comment:
Worth an explanatory comment? This looks puzzling without the context of the conversation we had in this PR.